### PR TITLE
fix: failing rdev updates

### PR DESCRIPTION
**Pull request by danieljhegeman**
## Reason for Change

- #5971 

## Changes

- Delete previous image version from ECR for a given PR/stack, for all ECR repos, after successful happy push of a new image version.
- Reintroduce unique sha-based image tags for all images.

## Testing steps

- Either list QA steps or reasoning you feel QA is unnecessary
- Describe how you made sure to know that your changes worked. Should allow someone else to go verify your code without in depth knowledge.
- "Unit tested only", "tested in rdev by a, b, c, verifying feature worked by... ", "manually ran pipeline locally with these results: ..."

### CODE REVIEW

File: .github/workflows/rdev-update-for-pr.yml [hunk 1]
```diff
@@ -56,6 +56,19 @@ jobs:
...
+          ECR_REPOS=(corpora-frontend corpora-backend corpora-upload-failures corpora-upload-success \
+            dataset-submissions corpora-upload wmg-processing cellguide-pipeline)
+          IMAGE_DIGESTS=()
+          for ecr_repo in "${ECR_REPOS[@]}"; do
+            IMAGE_DIGESTS+=(`aws ecr list-images --repository-name $ecr_repo --filter tagStatus=TAGGED \
+              | jq -r --arg STACK_NAME ${STACK_NAME} '[.imageIds[] \
+              | select(.imageTag==$STACK_NAME)][0].imageDigest'`)
+          done
+          echo "ECR_REPOS=${ECR_REPOS[@]}" >> $GITHUB_OUTPUT
+          echo "IMAGE_DIGESTS=${IMAGE_DIGESTS[@]}" >> $GITHUB_OUTPUT
```
Comment (danieljhegeman) [inline .github/workflows/rdev-update-for-pr.yml:71]: Does this code run each repo on each matrix slice? Yes. But that doesn't matter. It's quick, and whenever an image digest is missing, it just ignores that. Deletion attempts of images that don't exist aren't a problem.

Comment (danieljhegeman) [reply .github/workflows/rdev-update-for-pr.yml:71]: Can update to be a bit more streamlined later, but that will require separating out into a different job and configuring outputs. Might consider making it its own workflow later. Minor tech debt for another day.

Comment (danieljhegeman) [reply .github/workflows/rdev-update-for-pr.yml]: Nvm, went ahead and fixed this

Comment (danieljhegeman) [reply .github/workflows/rdev-update-for-pr.yml]: Nvm, went ahead and fixed this

**Review by ebezzi — APPROVED** [review]

File: .github/workflows/rdev-update-for-pr.yml [hunk 2]
```diff
@@ -28,6 +28,36 @@ permissions:
   pull-requests: write
 
 jobs:
+  get_previous_image_digests:
```
[the remark this answers is not in the document]
Comment (danieljhegeman) [reply .github/workflows/rdev-update-for-pr.yml:31]: the problem is [we need to access outputs by job name](https://github.com/chanzuckerberg/single-cell-data-portal/blob/aee6e26e935bc405598b5957dc93c10916aa9a9b/.github/workflows/rdev-update-for-pr.yml#L141)

[the remark this answers is not in the document]
Comment (ebezzi) [reply .github/workflows/rdev-update-for-pr.yml:31]: Ah, right. That makes sense!

**Pull request merged**